File: ProjectEuler088.c
#include <stdio.h>
int rec(int sum, int product, int k){
    if(sum <k)return 0;
    if(k==1&&sum==product)return 1;
    if(product==1&&sum==k)return 1;
    int i;
    for(i=2;i<=product&&(sum-i)>=k-1;++i)
        if(product%i==0&&rec(sum-i,product/i,k-1))return 1;
    return 0;
}
int main(){
    int i,j,sum=0;
    for(i=2;i<=12000;++i){
        for(j=i+1;!rec(j,j,i);++j);
        sum+=j;
    }
    printf("%d\n",sum);
    return 0;
}

File: ProjectEuler088.py
def rec(sums,product,k):
    if sums < k:return False
    if k==1 and product == sums: return True
    if product ==1 and k==sums: return True
    j=2
    while j <=product and sums-j>=k-1:
        if product %j==0 and rec(sums-j,product/j,k-1)==True:return True
        j+=1
    return False
tot=0
saved = []
for i in range(2,12001):
    j=i+1
    while rec(j,j,i)==False:
        j+=1
    if j not in saved:
        tot+=j
    saved.append(j)
print tot 
